qiskit/providers/experiment/device_component.py

```diff
@@ -0,0 +1,59 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+from abc import ABC, abstractmethod
+
+
+class DeviceComponent(ABC):
+    """Class representing a device component."""
+
+    @abstractmethod
+    def __str__(self):
+        pass
+
+    def __repr__(self):
+        return f"<{self.__class__.__name__}({str(self)})>"
+
+
+class Qubit(DeviceComponent):
```

Comment:
Note that a class named `Qubit` already exists in `qiskit/circuit/quantumregister.py`